### DISCUSSION TITLE
Export `Config` interface from `ziggy`

### DISCUSSION THREAD

**Discussion by apademide** [tighten/ziggy#799] [Ideas]
#### Description

In the context of an Inertia app, the Ziggy object is passed through the props under `$props.ziggy`. I could determine it is of the same format as the `Config` interface, which is not exposed. I found a way to extract the interface like that:
```ts
type ZiggyConfig = NonNullable<
  Parameters<typeof import("ziggy-js").useRoute>[0]
>;
```
which is pretty messy for such a simple type.

It would also not make sense to copy the type in my own codebase directly.

#### Suggestion

It'd be convenient to simply export the interface from Ziggy, `export interface Config { ... }`

**Comment by bakerkretzmar** [comment]
Where/how would you use that interface?

**Reply by apademide** [reply]
I extend Inertia `PageProps` to include props "globally reserved". This way, It's more convenient to work with the app as we know some props exists, and can't be used for anything else — and `ziggy` is one of them as it's set for all requests automatically. 

```tsx
import type { PageProps as InertiaPageProps } from "@inertiajs/core";

type ZiggyConfig = NonNullable<
  Parameters<typeof import("ziggy-js").useRoute>[0]
>;

/** Props that are standard across all pages. They might not be present in all pages, but must never be used as page-specific props holding different data. */
interface SharedPageProps {
  title?: string;
  ziggy?: ZiggyConfig;
  flash?: unknown;
  
  // ... rest of my global props
}

export type PageProps<T = {}> = InertiaPageProps & SharedPageProps & T;
```

**Reply by bakerkretzmar** [reply]
Thanks. Doesn't seem like it would be a problem to just export this on our side. That being said, if you're not using the contents of the `ziggy` prop Inertia is sending back I would highly recommend getting rid of it, there is some performance overhead (and if you are using it I would still actually suggest not doing so and putting `@routes` in your root Blade view instead).
